service-utils/gen-ssl.sh

```diff
@@ -1,49 +1,129 @@
-path_to_etc=/etc
-openssl_conf_path=/etc/ssl/ct-openssl.conf
+# Die on errors immediately.
+set -e
 
-echo "Setting up credential directory structure..."
-mkdir -p $path_to_etc/creds/root/certs
-mkdir -p $path_to_etc/creds/root/keys
-mkdir -p $path_to_etc/creds/services/certs
-mkdir -p $path_to_etc/creds/services/keys
+# First argument sets the location of /etc directory.
+path_to_etc=${1:?"An absolute path to /etc must be the first parameter."}
 
-mkdir tmp && cd tmp
+shift  # Cut off $1 so that the rest of $@ is a list of names.
+if [ -z "$*" ]; then
+    echo "No service names passed!"
+    exit 1
+fi
 
-echo "Generating root certs..."
-openssl genrsa -passout pass:1111 -des3 -out ca.key.pem 4096
+# Allow environment to override openssl config file.
+openssl_conf_file=${CT_OPENSSL_CONF_FILE:-/etc/ssl/ct-openssl.conf}
+if [ ! -f  "${openssl_conf_file}" ]; then
+    echo "No openssl config file at '${openssl_conf_file}'."
+    exit 1
+fi
```

Comment:
Parameter checks added. 
Setting `CT_OPENSSL_CONF_FILE` allows for easy (manual) local testing.